modules/dxp/apps/commerce/commerce-organization-web/src/main/resources/META-INF/resources/js/menu/MenuProvider.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * The contents of this file are subject to the terms of the Liferay Enterprise
+ * Subscription License ("License"). You may not use this file except in
+ * compliance with the License. You can obtain a copy of the License by
+ * contacting Liferay, Inc. See the License for the specific language governing
+ * permissions and limitations under the License, including but not limited to
+ * distribution rights of the Software.
+ */
+
+import ClayDropDown from '@clayui/drop-down';
+import React, {useEffect, useMemo, useRef, useState} from 'react';
+
+import AccountMenuContent from './AccountMenuContent';
+import OrganizationMenuContent from './OrganizationMenuContent';
+import UserMenuContent from './UserMenuContent';
+
+const content = {
+	account: AccountMenuContent,
+	organization: OrganizationMenuContent,
+	user: UserMenuContent,
+};
+
+export default function MenuProvider({
+	alignElementRef,
+	data: propData,
+	parentData,
+}) {
+	const [active, setActive] = useState(false);
+	const [data, updateData] = useState(false);
+	const menuRef = useRef(null);
+
+	/**
+	 * the useEffect below force the component repositioning
+	 */
+
+	useEffect(() => {
+		setActive(false);
+		updateData(propData);
+	}, [propData]);
+
+	useEffect(() => {
+		if (data) {
+			setActive(true);
+		}
+	}, [data]);
+
+	const MenuContent = useMemo(() => data && content[data.type], [data]);
+
+	function closeMenu() {
```

Comment:
Can We use () => (arrow functions) here?

Reply:
Yes! Is the hoisting the only reason why? (done it anyway)

Reply:
>Yes! Is the hoisting the only reason why? (done it anyway)

Yes, it's about hoisting and maintaining the same pattern as MenuContent that is another function